student-info-system/professor/views.py

```diff
@@ -96,10 +109,19 @@ def add_reference(request, sectionid):
                     messages.error(request,
                                    "There was a problem saving the new item to the database.")
                 return render(request, 'professor/reference_add.html', data)
-            section.refresh_reference_items()
+
+            # Specify all current+future sects by reg date or only current sections by reg date
+            if request.POST.get('semester_future') == 'future':
+                sects_to_update = course.section_set.exclude(
```

Comment:
Doesn't this imply that "future" includes current? As long as its clear to user. I mean, did they come here via *section* or *course*. If the former, makes sense. If from course, not sure.